### PR TITLE
New input option `pattern` added for DSL and metadata inputs

### PR DESCRIPTION
**Pull request by Nik08**
## Description

1. New input option `pattern` added for DSL and metadata inputs
2. Bug fixes for other input options in DSL like, `required`, `type` and `description`

## Related Issue

Limited feature #5014  

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by Vasu1105** [comment]
@Nik08 I think that test is failing as you need to add the newly added field in the to_hash method of lib/inspec/objects/input.rb here https://github.com/inspec/inspec/blob/master/lib/inspec/objects/input.rb#L23 or here too https://github.com/inspec/inspec/blob/master/lib/inspec/input.rb#L327

**Comment by Nik08** [comment]
@Vasu1105 Nice catch! I will try this and see if it resolves the build issue.

**Review by clintoncwolfe — APPROVED** [review]
Great work!

File: docs-chef-io/content/inspec/inputs.md [hunk 1]
```diff
@@ -416,6 +416,13 @@ input values that are used as test results.
 
 Allowed in: Metadata
 
+### Pattern
+
+Optional, `Regexp`. It should have a value of a regular expression. This feature helps
+in validating input value by matching it with the provided regular expression.
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/inputs.md]:
```suggestion
Optional, `Regexp`. This feature validates the input by matching it with the provided regular expression.
```

**Pull request merged**